clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -2311,6 +2311,38 @@ static mlir::Value createLLVMBitOp(mlir::Location loc,
   return result;
 }
 
+class CIRObjSizeOpLowering
+    : public mlir::OpConversionPattern<mlir::cir::ObjSizeOp> {
+public:
+  using OpConversionPattern<mlir::cir::ObjSizeOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::ObjSizeOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    auto resTy =
+        getTypeConverter()->convertType(op.getType()).cast<mlir::IntegerType>();
+    auto loc = op->getLoc();
+
+    std::string llvmIntrinName = "llvm.objectsize";
+    auto llvmIntrinNameAttr =
+        mlir::StringAttr::get(rewriter.getContext(), llvmIntrinName);
+    // kindInfo can only be 0 or 2, a.k.a min or max,
+    // should we have a validity check here?
+    mlir::cir::SizeInfoType kindInfo = op.getKind();
+    auto falseValue = rewriter.create<mlir::LLVM::ConstantOp>(
+        loc, rewriter.getI1Type(), false);
+    auto trueValue = rewriter.create<mlir::LLVM::ConstantOp>(loc, rewriter.getI1Type(), true);
+
+    rewriter.replaceOpWithNewOp<mlir::LLVM::CallIntrinsicOp>(
+        op, resTy, llvmIntrinNameAttr, mlir::ValueRange{adaptor.getPtr(),
+                                                        kindInfo ==  mlir::cir::SizeInfoType::max?falseValue:trueValue,
```

Comment:
This is not properly formatted, you'll likely get a bot message soon. I suggest you use clang-format before submitting PRs: https://clang.llvm.org/docs/ClangFormat.html